src/dc_federated/backend/zmq_interface.py

```diff
@@ -0,0 +1,107 @@
+"""
+Two classes for defining the ZeroMQ interface between the server and the model.
+"""
+
+import zmq
+import logging
+
+logger = logging.getLogger(__name__)
+logger.setLevel(level=logging.INFO)
+
+
+class ZMQInterfaceModel:
```

Comment:
I wanted to keep the two sides of the interface in the same file so that you could easily see the reflexive relationship between them. If another function is added to the interface it is then more clear where you need to update the other part of the interface.

Reply:
Yeah this is a good idea I'll do this 👍